### PR TITLE
Iterables

### PR DESCRIPTION
**Pull request by mahdiHashemi14**
Please review and comment / commit if anything is wrong

### CODE REVIEW

File: 1-js/05-data-types/06-iterable/article.md [hunk 1]
```diff
@@ -1,58 +1,58 @@
...
-2. Onward, `for..of` works *only with that returned object*.
-3. When `for..of` wants the next value, it calls `next()` on that object.
-4. The result of `next()` must have the form `{done: Boolean, value: any}`, where `done=true`  means that the iteration is finished, otherwise `value` is the next value.
+1. زمانی که `for..of` شروع می‌شود، متد را یک بار صدا می‌زند (یا اگر پیدا نشود ارور می‌دهد). متد باید یک *حلقه‌زننده* را برگرداند -- شیءای که متد `next` را دارد.
+2. همینطور رو به جلو، `for..of` *تنها با شیء برگردانده شده* کار می‌کند.
+3. زمانی که `for..of` مقدار بعدی را نیاز دارد، روی آن شیء `next()` را صدا می‌زند.
+4. نتیجه `next()` باید به شکل `{done: Boolean, value: any}` باشد که `done=true` به معنی پایان حلقه‌زدن است، در غیر این صورت `value` مقدار بعدی خواهد بود.
 
-Here's the full implementation for `range` with remarks:
+اینجا پیاده‌سازی کامل را برای `range` به همراه اظهارات داریم:
```
Comment (mahdyar) [inline 1-js/05-data-types/06-iterable/article.md:36]:
```suggestion
اینجا پیاده‌سازی کامل `range` را به همراه ملاحظات داریم:
```

File: 1-js/05-data-types/06-iterable/article.md [hunk 2]
```diff
@@ -1,58 +1,58 @@
...
 
 ```js run
 let range = {
   from: 1,
   to: 5
 };
 
-// 1. call to for..of initially calls this
+// 1. ابتدا این متد صدا زده می‌شود for..of با استفاده از
```
Comment (mahdyar) [inline 1-js/05-data-types/06-iterable/article.md:44]:
```suggestion
// 1. در ابتدا این متد صدا زده می‌شود for..of با صدازدن
```

File: 1-js/05-data-types/06-iterable/article.md [hunk 3]
```diff
@@ -211,68 +211,68 @@ let arrayLike = {
...
 alert(chars[0]); // 𝒳
 alert(chars[1]); // 😂
 alert(chars.length); // 2
 ```
 
-Unlike `str.split`, it relies on the iterable nature of the string and so, just like `for..of`, correctly works with surrogate pairs.
+برخلاف `str.split`، این روش بر اساس طبیعت حلقه‌پذیری رشته کار می‌کند و به همین دلیل، درست مانند `for..of`، با جفت‌های جایگیر به درستی کار می‌کند.
 
-Technically here it does the same as:
+از لحاظ فنی اینجا هم کار مشابهی را انجام می‌دهد:
```
Comment (mahdyar) [inline 1-js/05-data-types/06-iterable/article.md:260]:
```suggestion
از لحاظ فنی این هم کار مشابهی را انجام می‌دهد:
```

File: 1-js/05-data-types/06-iterable/article.md [hunk 4]
```diff
@@ -211,68 +211,68 @@ let arrayLike = {
...
+let chars = []; // هم از درون این حلقه را اجرا می‌کند Array.from
 for (let char of str) {
   chars.push(char);
 }
 
 alert(chars);
 ```
 
-...But it is shorter.    
+...اما این کوتاه‌تر است.
```
Comment (mahdyar) [inline 1-js/05-data-types/06-iterable/article.md:273]:
```suggestion
...اما این روش کوتاه‌تر است.
```

**Review by mahdyar — CHANGES_REQUESTED** [review]

**Comment by mahdiHashemi14** [comment]
Thank you, @mahdyar 
/done

**Review by mahdyar — APPROVED** [review]
LGTM!
Thank you, @mahdiHashemi14!

**Pull request merged**